src/checkbox/tests/unit/Checkbox.ts

```diff
@@ -1,151 +1,230 @@
-		const vnode = <VNode> checkbox.__render__();
-		const inputNode = vnode.children![0].children![0];
+	simple() {
+		const children: HNode[] = [
+			v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				null,
+				null,
+				v('input', {
+					'aria-describedby': undefined,
+					'aria-invalid': null,
+					'aria-readonly': null,
+					bind: widget,
+					checked: false,
+					classes: widget.classes(css.input),
+					disabled: undefined,
+					name: undefined,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: undefined,
+					required: undefined,
+					type: 'checkbox',
+					value: undefined
+				})
+			])
+		];
+
+		widget.setChildren(children);
 
-		assert.strictEqual(inputNode.vnodeSelector, 'input');
-		assert.strictEqual(inputNode.properties!.type, 'checkbox');
-		assert.strictEqual(inputNode.properties!.checked, false);
+		widget.expectRender(v('div', {
+			classes: widget.classes(css.root)
+		}, children));
 	},
 
-	'correct node attributes'() {
-		const checkbox = new Checkbox();
-		checkbox.setProperties({
+	'properties and attributes'() {
+		const checkboxProperties: CheckboxProperties = {
+			bind: widget,
 			checked: true,
 			describedBy: 'id1',
 			disabled: true,
 			formId: 'id2',
 			invalid: true,
 			label: 'foo',
+			mode: Mode.toggle,
 			name: 'bar',
+			offLabel: 'Off',
+			onLabel: 'On',
 			readOnly: true,
 			required: true,
 			value: 'qux'
-		});
-		const vnode = <VNode> checkbox.__render__();
-		const labelNode = vnode.children![0];
-		const inputNode = vnode.children![1].children![0];
-
-		assert.isTrue(inputNode.properties!.checked);
-		assert.strictEqual(inputNode.properties!['aria-describedby'], 'id1');
-		assert.isTrue(inputNode.properties!.disabled);
-		assert.strictEqual(inputNode.properties!['aria-invalid'], 'true');
-		assert.strictEqual(inputNode.properties!.name, 'bar');
-		assert.isTrue(inputNode.properties!.readOnly);
-		assert.strictEqual(inputNode.properties!['aria-readonly'], 'true');
-		assert.isTrue(inputNode.properties!.required);
-		assert.strictEqual(inputNode.properties!.value, 'qux');
-
-		assert.strictEqual(vnode.properties!['form'], 'id2');
-		assert.strictEqual(labelNode.properties!.innerHTML, 'foo');
-	},
+		};
+		const children: HNode[] = [
+			v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				v('div', { classes: widget.classes(css.onLabel) }, [ checkboxProperties.onLabel! ]),
+				v('div', { classes: widget.classes(css.offLabel) }, [ checkboxProperties.offLabel! ]),
+				v('input', {
+					'aria-describedby': checkboxProperties.describedBy,
+					'aria-invalid': String(checkboxProperties.invalid),
+					'aria-readonly': String(checkboxProperties.readOnly),
+					bind: widget,
```

Comment:
Hm, maybe it's only `assertRender` then (https://github.com/dojo/test-extras/blob/master/src/support/assertRender.ts#L40)? @agubler would know better than me :)